qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/QuteCodeActions.java

```diff
@@ -169,6 +178,40 @@ public CompletableFuture<List<CodeAction>> doCodeActions(Template template, Code
 					// Do nothing
 				}
 			}
+			QuteSyntaxErrorCode syntaxErrorCode = QuteSyntaxErrorCode.getErrorCode(diagnostic.getCode());
+			if (syntaxErrorCode != null) {
+				try {
+					CodeActionRequest request = new CodeActionRequest(template, range.getEnd(), diagnostic,
+							javaTextEditProvider, sharedSettings);
+					switch (syntaxErrorCode) {
+					case UNTERMINATED_SECTION:
+						// Given the following Qute template
+						// {#if variable}
+						//
+						// Provide a code action to add an end section tag
+						// {#if variable}
+						// {/if}
+						codeActionForUnterminatedSection.doCodeActions(request, codeActionResolveFutures, codeActions);
+						break;
+					case SECTION_END_DOES_NOT_MATCH_START:

```

Comment:
I think you could add `case SECTION_BLOCK_END_DOES_NOT_MATCH_START` here (no tested)

Reply:
Ok thanks for yoru feedback. It is more complicated that I though. We can do that in a separate PR. Please create an issue for that.

Reply:
I have created an issue here: https://github.com/redhat-developer/quarkus-ls/issues/844